doc/design/executors.md

```diff
@@ -0,0 +1,187 @@
+  ascribe the just-run actor's reductions to itself in its own load statistics, but the recieving
+  executor will. Doing so causes the change in load to show up almost immediately in load
+  statistics. The receiving executor schedules the actor `N` segments forward.
+- (Jason and I mentioned finding an actor somewhere in the wheel that we'd expect to run around the
+  same time as the end of the receiving executor's runnable tail).
+- Idle actors that have just become runnable. The executor will inspect N other random executors to
+  compare load statistics. Based on the comparison, the executor may migrate the actor.
+
+Actors that may not be migrated.
+- Posse members, at least not to executors that are part of the cabal.
+- Actors marked as pinned by their supervisor startup strategy.
+- Idle actors waiting on mailbox completions. Doing so would cause degenerate load balancing
+  overhead long before the idle actors contribute to meaningful load. Additionally, in the case
+  where the idle actor is waiting on I/O completion messages, it is optimal to keep such actors
+  associated with the executor with which they submitted I/O. It has the upside of reducing
+  inter-executor I/O completion messaging. Migrating such actors would guarantee at least one
+  follow-up message for the I/O completion.
+
+If idle actors become active, normal load balancing mechanisms kick in to migrate based on current,
+non-speculative load statistics.
+
+## I/O cancellation
+
+## Supervisors
+### Strategies
+#### Graph
+
+DAG of actor dependencies that determines startup order.
+
+#### CPU selection
+
+- Pinned
+- NUMA groups
+- Big/little CPU
+
+#### Crash recovery
+- pattern
+   - one
+   - upstream
+   - downstream
+   - all
+- times
+   - n
+   - n in t time window
+
+## Actors
+
+### Time slice procedures
+
+Between each time slice, the executor is responsible for the following.
+1. Reap any CQEs from the ioring's completion queue. Determine which actor is associated with the
+   CQE. Copy the CQE to the location pointed to by the CQE's `user_data` field. If the actor is...
+   a. Not managed by the executor - send a message to the managing executor with a hint that the
+   actor should be activated.
+   b. `Receiving` - move the actor from the `Receiving` list to the `Runnable` queue.
+2. If actor placed any SQEs in the ioring, submit it via `io_uring_submit`.
+3. If the actor state is...
+   a. `Receiving` - place it at the head of the idle actor list.
+   b. `Runnable` - place it at the tail of the active actor queue.
+   c. `Halted` - place a message in the actor's supervisor's mailbox and place the actor in the halted actor list
+
+### Supervisors
+
+### States
+- Waiting - Waiting for messages in the mailbox.
+- Runnable - In the `Runnable` queue, waiting to be run by the executor.
+- Running - Actively running.
+- Halted - Halted and requires cleanup and possibly supervisor actions.
```

Comment:
It looks like `Receiving` and `Waiting` are identical? Also, the "States" subsection should come before "Time slice procedures" so that the definitions are stated before first use.